test/input-states-mocha.js

```diff
@@ -0,0 +1,341 @@
+/**
+ * File: input-state-mocha.js
+ * Unit tests for the input-state APIs defined in src/input-state.js
+ */
+
+var chai = require('chai');
+var expect = chai.expect;
+var should = chai.should();
+
+var _ = require('underscore');
+var componentFactory = require('../src/noflo-component-factory');
+var access = require('../src/noflo-component-access');
+var test = require('./common-test');
+var inputStates = require('../src/input-states');
+var createLm = require('../src/create-lm');
+
+describe('input-states', function() {
+    var component, network;
+    var oninput, onoutput;
+    beforeEach(function(){
+        var getComponent = function() {
+            var factory = componentFactory({
+                inPorts:{
+                    input:{
+                        ondata: function(payload) {
+                            oninput(payload);
```

Comment:
I would have thought that our componentFactory would already set the ondata function, so it would not have to be set explicitly here.  Maybe that is not implemented yet?